tests/test_patron.py

```diff
@@ -0,0 +1,308 @@
+@freeze_time("2023-03-01")
+def test_patron_xml_from_records_student_success():
+    student_patron_all_values_record_1 = (
+        "111111111",
+        "STUDENT_KRB_NAME@MIT.EDU",
+        "STUDENT_KRB_NAME",
+        "11111111111111",
+        "Doe",
+        "Jane",
+        "Janeth",
+        "100 Smith St",
+        "Apt 34",
+        "Cambridge",
+        "MA",
+        "00000",
+        "5555555555",
+        "4444444444",
+        "3333333333",
+        "G",
+        "1",
+    )
+    student_patron_all_values_record_2 = (
+        "333333333",
+        "STUDENT_KRB_NAME@MIT.EDU",
+        "STUDENT_KRB_NAME",
+        "33333333333333",
+        "Doe",
+        "Jane",
+        "Janeth",
+        "100 Smith St",
+        "Apt 34",
+        "Cambridge",
+        "MA",
+        "00000",
+        "5555555555",
+        "4444444444",
+        "3333333333",
+        "G",
+        "1",
+    )
+    with open(
+        "tests/fixtures/student_patron_xml_record_1.xml", "r", encoding="utf8"
+    ) as xml_file_1, open(
+        "tests/fixtures/student_patron_xml_record_2.xml", "r", encoding="utf8"
+    ) as xml_file_2:
+        student_patron_record_1 = BeautifulSoup(
+            xml_file_1.read(),
+            features="xml",
+        )
+        student_patron_record_2 = BeautifulSoup(
+            xml_file_2.read(),
+            features="xml",
+        )
+        patron_xml_records = patron_xml_from_records(
+            "student",
+            [student_patron_all_values_record_1, student_patron_all_values_record_2],
+        )
+        assert next(patron_xml_records) == student_patron_record_1
+        assert next(patron_xml_records) == student_patron_record_2
+
+
+def test_populate_patron_common_fields_staff_all_values_success(
+    staff_patron_template, staff_patron_all_values_dict
+):
+    patron_xml_record = populate_patron_common_fields(
+        staff_patron_template,
+        staff_patron_all_values_dict,
+        SIX_MONTHS,
+        TWO_YEARS,
+    )
+    assert patron_xml_record.primary_id.string == "STAFF_KRB_NAME@MIT.EDU"
+    assert patron_xml_record.expiry_date.string == "2023-09-01Z"
+    assert patron_xml_record.purge_date.string == "2025-09-01Z"
+    assert (
+        patron_xml_record.emails.email.email_address.string == "STAFF_KRB_NAME@MIT.EDU"
+    )
+    assert (
+        patron_xml_record.user_identifiers.user_identifier.value.string == "222222222"
+    )
+    assert (
+        patron_xml_record.find_all("user_identifier")[1].value.string
+        == "22222222222222"
+    )
+
+
+def test_populate_patron_common_fields_staff_no_krb_name_but_email_success(
+    staff_patron_template, staff_patron_all_values_dict
+):
+    staff_patron_all_values_dict["KRB_NAME_UPPERCASE"] = None
+    patron_xml_record = populate_patron_common_fields(
+        staff_patron_template,
+        staff_patron_all_values_dict,
+        SIX_MONTHS,
+        TWO_YEARS,
+    )
+    assert patron_xml_record.primary_id.string == "STAFF_KRB_NAME@MIT.EDU"
```

Comment:
One thing you could do to make this test clearer is have the email address be different from the kerb name in the input record so it's obvious where the value came from.

Reply:
While looking at the reject functionality in the original script, I found this bit of code doesn't do anything because any record without a kerb goes to the reject file, I'm removing the code and tests